classes/class-wc-connect-compatibility-wc30.php

```diff
@@ -130,6 +130,33 @@ public function get_product_name_from_order( $product_id, $order ) {
 			return sprintf( __( '#%d - [Deleted product]', 'woocommerce-services' ), $product_id );
 		}
 
+		/**
+		 * For a given product ID, it tries to find its price inside an order's line items.
+		 *
+		 * @param int $product_id Product ID or variation ID
+		 * @param WC_Order $order
+		 * @return float The product (or variation) price, or NULL if it wasn't found
```

Comment:
What are the circumstances in which the product price is not found?

Reply:
If the item was removed from the order, for example. Super-edge case, but it's a possibility.

Reply:
Doesn't line_items have the latest items in the order?